src/session/session.ts

```diff
@@ -0,0 +1,83 @@
+import {IMap} from "./types";
+import * as crypto from "crypto";
+import {COOKIE_SECRET, DEFAULT_SESSION_EXPIRATION} from "./config";
+import * as keys from "./keys";
+
+export default class Session {
+    get cookieId(): string {
+        return this._cookieId;
+    }
+
+    set data(data: IMap<any>) {
+        this._data = data;
+    }
+
+    get data(): IMap<any> {
+        return this._data;
+    }
+
+    /**
+     * Static factory to create a new session instance using the default store
+     * (redis).
+     */
+    public static newInstance(): Session {
+        const session = new Session();
+        const now = Math.floor((new Date()).getTime() / 1000);
+        session.data = {
+            [keys.ID]: session.sessionKey(),
+            [keys.EXPIRES]: now + parseInt(DEFAULT_SESSION_EXPIRATION, 10),
```

Comment:
This '10' is a magic number! Would be good to define a constant somewhere that describes what it actually is (number_of_minutes_before_session_expiry?) and reference it here.